Remove borland support

thomasdziedzic opened the pull request:
See https://github.com/vim/vim/issues/3374

thomasdziedzic committed:
remove references to bc5

thomasdziedzic committed:
get rid of all borland references

tonymec commented:
* `runtime/compiler/bcc.vim` is for running bcc to compile **anything** (not only Vim) with Vim's quickfix feature. Should we also get rid of it?
* `runtime/doc/usr90.txt`: the proposed change gets rid of the verb in one sentence.
* `src/dosinst.c`: If we get rid of bc5 (Borland C version 5), shouldn't we *a fortiori* get rid of bc3 (Borland C version 3)?
* `src/gui_w32.c`, new line 5845: there is now an extraneous comma in `for Cygwin or MinGW`.
* `src/vim.rc`: Are you sure the `#else` branch (`#include <winres.h>`) is not used by e.g. MSVC?

Best regards,
Tony.

thomasdziedzic committed:
add back bcc.vim runtime file

thomasdziedzic committed:
fix edited sentence in usr_90.txt

thomasdziedzic committed:
remove reference to bc3

thomasdziedzic committed:
remove extraneous comma

thomasdziedzic commented:
> runtime/compiler/bcc.vim is for running bcc to compile anything (not only Vim) with Vim's quickfix feature. Should we also get rid of it?

I think I will add it back in. I wanted to only get rid of the bits related to compiling vim with borland. I guess vim can still be used to edit with the borland compiler though.

> runtime/doc/usr90.txt: the proposed change gets rid of the verb in one sentence.

good catch

> src/dosinst.c: If we get rid of bc5 (Borland C version 5), shouldn't we a fortiori get rid of bc3 (Borland C version 3)?

I think bc3 support already doesn't exist. That just looks like some outdated info, and I couldn't find any files named bc3 in the repo. I updated the docs.

> src/gui_w32.c, new line 5845: there is now an extraneous comma in for Cygwin or MinGW.

fixed

> src/vim.rc: Are you sure the #else branch (#include <winres.h>) is not used by e.g. MSVC?

Those are all commented out, so I removed the dead code completely.

Thanks for your review @tonymec

k-takata commented:
Two years ago, Embarcadero C++ 10.1 was released. At that time, I was able to build Vim with it with a small changes to the Make_bc5.mak:
https://bitbucket.org/k_takata/vim-ktakata-mq/src/3244815cd8cac1fd7e8379de244880e8b481d20c/embarcadero.patch?at=default&fileviewer=file-view-default
However, I try it again today, it cannot be built anymore. It's obvious that no one cared Borland (Embarcadero) since then.
I think it's good time to drop support for Borland.
